### PR TITLE
Adds os guesser tests

### PR DESCRIPTION
**Pull request by mdaif**
adds OS detection methods for zte, isam, and vxworks along with banner files.

### CODE REVIEW

**Comment by egroeper** [comment]
Thanks. 
Sorry, I don't have time at the moment. I will have a look at it as soon as possible.

File: src/Exscript/protocols/drivers/alcatel.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+_user_re     = [re.compile(r'login: ?', re.I)]
+_password_re = [re.compile(r'[\r\n]password: ?', re.I)]
+_prompt_re   = [re.compile(r'[\r\n][\- +\d+\w+\.]+(?:\([^\)]+\))?[>#] ?')]
+_error_re    = [re.compile(r'%Error'),
+                re.compile(r'invalid token', re.I),
+                re.compile(r"command is not complete")]
+
+class IsamDriver(Driver):
+    def __init__(self):
+        Driver.__init__(self, 'Isam')
```
Comment (egroeper) [inline src/Exscript/protocols/drivers/alcatel.py]: driver name needs to match the driver (alcatel)

File: src/Exscript/protocols/drivers/isam.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class IsamDriver(Driver):
+    def __init__(self):
+        Driver.__init__(self, 'isam')
+        self.user_re     = _user_re
+        self.password_re = _password_re
+        self.prompt_re   = _prompt_re
+        self._error_re   = _error_re
+
+    def check_response_for_os(self, string):
+        if
```
Comment (egroeper) [inline src/Exscript/protocols/drivers/isam.py]: invalid line

**Comment by egroeper** [comment]
Do you have a banner file for the alcatel driver, too?

**Comment by egroeper** [comment]
Unfortunately your vxworks banner gets detected by the existing vrp driver, which makes the os guesser tests fail.
Do you have know what the differences between vrp and vxworks are? It seems, that at least some of the vrp versions are derived of vxworks.
I'm unsure what the best fix is here.
Does the vrp driver work for your MA5600?

**Comment by mdaif** [comment]
I thought I pushed some modifications but I didn't .. sorry for that :$
I renamed alcatel to isam to reflect the devices I tested my code against. No the vrp driver didn't work for my MA5600, that's why we requested the specific OS name from Huawei.
I will push the modifications then I'll make a new pull request. Everything should be fine as I ran the test cases locally and they all pass :)

**Comment by mdaif** [comment]
hi .. did you have the time to review the latest two commits ? (I apologize for the inconvenience but this is my first contribution to an opensource project :) )

**Comment by egroeper** [comment]
Sorry. Your last comment stated 
> I will push the modifications then I'll make a new pull request

so I waited (did nothing) for it. Will have a look at it soon.
Is it complete now and passes the tests locally?

**Comment by mdaif** [comment]
yes ... all tests are passing locally now :)

**Comment by egroeper** [comment]
Sorry, that it took me so long.
Thanks for your contribution.

**Pull request merged**